FWCore/Framework/interface/EventSetupRecordProvider.h

```diff
@@ -102,43 +124,51 @@ namespace edm {
       std::shared_ptr<EventSetupRecordIntervalFinder>& finder() { return get_underlying_safe(finder_); }
 
       void getReferencedESProducers(
-          std::map<EventSetupRecordKey, std::vector<ComponentDescription const*>>& referencedESProducers);
+          std::map<EventSetupRecordKey, std::vector<ComponentDescription const*>>& referencedESProducers) const;
 
       void fillReferencedDataKeys(std::map<DataKey, ComponentDescription const*>& referencedDataKeys) const;
 
       void resetRecordToProxyPointers(DataToPreferredProviderMap const& iMap);
 
+      void setEventSetupImpl(EventSetupImpl* value) { eventSetupImpl_ = value; }
```

Comment:
The EventSetupRecordProvider has a vector of EventSetupRecordImpl, one for each of the possible concurrent IOVs. EventSetupRecordProvider uses this pointer to the EventSetupImpl to tell it which of those EventSetupRecordImpl's to use when the IOV starts to be processed. Note that eventSetupImpl_ points to the most recent EventSetupImpl and this only works because we know the next EventSetupImpl cannot be created until after eventSetupImpl_ has been used and is not needed anymore. We do not need to keep a pointer to all the EventSetupImpls's in flight, just the most recent one. In EventSetupRecordProvider see functions initializeForNewIOV and continueIOV.